rest/data/commit_queue.go

```diff
@@ -577,3 +580,87 @@ func GetAdditionalPatches(patchId string) ([]string, error) {
 		Message:    errors.Errorf("patch '%s' not found in commit queue", patchId).Error(),
 	}
 }
+
+// CheckCanRemoveCommitQueueItem checks if a patch can be removed from the commit queue by the given user.
+func CheckCanRemoveCommitQueueItem(ctx context.Context, sc Connector, usr *user.DBUser, project *model.ProjectRef, itemId string) error {
+	if !project.CommitQueue.IsEnabled() {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    errors.Errorf("commit queue is not enabled for project '%s'", project.Id).Error(),
+		}
+	}
+
+	if canAlwaysSubmitPatchesForProject(usr, project.Id) {
+		return nil
+	}
+
+	if bson.IsObjectIdHex(itemId) {
+		patch, err := FindPatchById(itemId)
+		if err != nil {
+			return gimlet.ErrorResponse{
+				StatusCode: http.StatusInternalServerError,
+				Message:    errors.Wrapf(err, "finding patch '%s'", itemId).Error(),
+			}
+		}
+		if usr.Id != utility.FromStringPtr(patch.Author) {
+			return gimlet.ErrorResponse{
+				StatusCode: http.StatusUnauthorized,
+				Message:    "not authorized to patch on behalf of author",
```

Comment:
Sorry to only catch this late, but this error message (and the Github one below) seem wrong; they may be outdated from when this check was only for patching? I'm not sure. It would probably more accurate to say "not authorized to perform action on behalf of author" since this is used in a few places.

Reply:
makes sense, I will update this text!